tensorboard/plugins/core/core_plugin_test.py

```diff
@@ -133,8 +134,10 @@ def testLogdir(self):
 
   def testRuns(self):
     """Test the format of the /data/runs endpoint."""
-    run_json = self._get_json(self.logdir_based_server, '/data/runs')
-    self.assertEqual(run_json, ['run1'])
+    # Test both database and logdir mode.
+    for server in (self.db_based_server, self.logdir_based_server):
+      self.assertEqual(['run1'], self._get_json(server, '/data/runs'))
+
 
   def testRunsAppendOnly(self):
```

Comment:
Unfortunately, I reverted this test because it threw this error both locally and on travis.

```
rboard/plugins/core:core_plugin_test
    Testing //tensorboard/plugins/core:core_plugin_test; 4s linux-sandbox
    Testing //tensorboard/plugins/graph:graphs_plugin_test; 1s linux-sandbox
e='run1')
  File "/home/travis/.bazel-output-base/bazel-sandbox/4995855409685118630/execroot/org_tensorflow_tensorboard/bazel-out/k8-fastbuild/bin/tensorboard/plugins/core/core_plugin_test.runfiles/org_tensorflow_tensorboard/tensorboard/plugins/core/core_plugin_test.py", line 242, in _generate_test_data
    user_name='user').as_default() as db_writer:
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/tensorflow/contrib/summary/summary_ops.py", line 265, in create_db_writer
    user_name=user_name)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/tensorflow/contrib/summary/summary_ops.py", line 275, in _make_summary_writer
    factory(resource, **kwargs))
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/tensorflow/contrib/summary/gen_summary_ops.py", line 118, in create_summary_db_writer
    user_name=user_name, name=name)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 3259, in create_op
    op_def=op_def)
  File "/home/travis/virtualenv/python2.7.14/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1650, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access
InvalidArgumentError (see above for traceback): Sqlite::Open(sqlite:/home/travis/.bazel-output-base/bazel-sandbox/4995855409685118630/execroot/org_tensorflow_tensorboard/tmp/core_plugin_testjzahzF/tmpTGlfAJ/db.db) failed: unable to open database file
	 [[Node: CreateSummaryDbWriter_18 = CreateSummaryDbWriter[_device="/job:localhost/replica:0/task:0/device:CPU:0"](SummaryWriter_18, CreateSummaryDbWriter_18/db_uri, Const_36, Const_37, Const_38)]]
```

It seems like writing to the database from the test does not work. Using an in-memory database does not work either because the connection is closed after the DB writer closes, at which time the database is wiped from memory.

Unit-testing SQL logic might require some more planning.

Reply:
I'm trying to manually insert the rows for the runs ... we'll need a solution that scales better, preferably using the DB writer, for future tests.

Reply:
Ah, it seems like I can't write to the database in general from the test.

Reply:
Indeed, setting up the schema works, albeit it is missing a few fields:

```
CREATE TABLE Runs (
          rowid INTEGER PRIMARY KEY,
          run_id INTEGER NOT NULL,
          experiment_id INTEGER NOT NULL,
          run_name VARCHAR(1900) NOT NULL
        );
```

I wonder if that is part of it. `create_file_writer` runs fine.

Reply:
Issue fixed. See https://github.com/tensorflow/tensorboard/pull/985/commits/7791e75afaf8226b4c7d48b109adbbd9ae593b3d. PTAL :)